tests/test_integration.py

```diff
@@ -249,9 +279,28 @@ def test_rebuild(self):
             vm.run('touch /root/initial_canary')
             vm.run('test -f /root/initial_canary')
 
-        # Now stop it and rebuild it
         vm_stop(VM_HOSTNAME)
-        vm_build(VM_HOSTNAME, rebuild=True)
+
+        if self.datacenter_type == 'kvm.dct':
+            # Now stop it and rebuild it
+            vm_build(VM_HOSTNAME, rebuild=True)
+        elif self.datacenter_type == 'aws.dct':
+            vm_delete(VM_HOSTNAME, retire=True)
```

Comment:
same here. we should try to develop clear and consistent apis. i would expect `vm_build(VM_HOSTNAME, rebuild=True)` to do exactly the same thing on aws that it would do in our own datacenter. the `vm_build()` function should internally decide what to do based on the `datacenter_type` and not the test :)

Reply:
reworked